plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPreferencesPanel.java

```diff
@@ -0,0 +1,411 @@
+    } catch (BackingStoreException exception) {
+      logger.log(Level.SEVERE, "Could not save deploy preferences", exception);
+      MessageDialog.openError(getShell(),
+                              Messages.getString("deploy.preferences.save.error.title"),
+                              Messages.getString("deploy.preferences.save.error.message",
+                                                 exception.getLocalizedMessage()));
+      return false;
+    }
+  }
+
+  private void loadPreferences(IProject project) {
+    model = new DeployPreferencesModel(project);
+  }
+
+  private void createProjectIdSection() {
+    Composite projectIdComp = new Composite(this, SWT.NONE);
+    projectIdComp.setLayout(new GridLayout(2, false));
+    projectIdComp.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+
+    projectIdLabel = new Label(projectIdComp, SWT.LEFT);
+    projectIdLabel.setLayoutData(new GridData(SWT.LEFT, SWT.CENTER, false, false));
+    projectIdLabel.setText(Messages.getString("project.id"));
+
+    projectId = new Text(projectIdComp, SWT.LEFT | SWT.SINGLE | SWT.BORDER);
+    projectId.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+  }
+
+  private void createProjectVersionSection() {
+    Composite versionComp = new Composite(this, SWT.NONE);
+    versionComp.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+    versionComp.setLayout(new GridLayout(2, false));
+
+    overrideDefaultVersionButton = new Button(versionComp, SWT.CHECK);
+    overrideDefaultVersionButton.setText(Messages.getString("use.custom.versioning"));
+    overrideDefaultVersionButton.setLayoutData(new GridData(SWT.LEFT, SWT.CENTER, true, false, 2, 1));
+
+    versionLabel = new Label(versionComp, SWT.NONE);
+    versionLabel.setText(Messages.getString("project.version"));
+    GridData layoutData = new GridData(SWT.LEFT, SWT.CENTER, false, false);
+    layoutData.horizontalIndent = INDENT_CHECKBOX_ENABLED_WIDGET;
+    versionLabel.setLayoutData(layoutData);
+
+    version = new Text(versionComp, SWT.LEFT | SWT.SINGLE | SWT.BORDER);
+    version.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+  }
+
+  private void createPromoteSection() {
+    Composite promoteComp = new Composite(this, SWT.NONE);
+    promoteComp.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+    promoteComp.setLayout(new GridLayout(1, false));
+    autoPromoteButton = new Button(promoteComp, SWT.CHECK);
+    autoPromoteButton.setText(Messages.getString("auto.promote"));
+    autoPromoteButton.setLayoutData(new GridData(SWT.LEFT, SWT.CENTER, true, false));
+
+    Link manualPromoteLink = new Link(promoteComp, SWT.NONE);
+    GridData layoutData = new GridData(SWT.LEFT, SWT.CENTER, true, false);
+    layoutData.horizontalIndent = INDENT_CHECKBOX_ENABLED_WIDGET;
+    manualPromoteLink.setLayoutData(layoutData);
+    manualPromoteLink.setText(Messages.getString("deploy.manual.link", APPENGINE_DASHBOARD_URL));
```

Comment:
I remember now why I decided to send the user to the App Engine dashboard: if they have worked with a different project previously, the web console will bring that up by default, not the current project. But I agree it's closer to where the user should be, also if we have the project ID we can add it to the URL as you suggested.